### ISSUE TITLE
Roll up errors in CoVE for IATI

### ISSUE THREAD

**Issue by robredpath** [OpenDataServices/cove#899]
In cases where there's lots of the same error for IATI rulesets, we'd like to roll them up and present them in a nicer way. 

We'll look at some example data in deciding the best way to do this.

**Comment by robredpath** [comment]
@rory09 @edugomez this might be a good thing for you to start to look at in the office this week

**Comment by edugomez** [comment]
@rory09 and I had a discussion on this issue. We decided that one of the things we need to do to structure ruleset results better is to change the terminology used:

- What we call 'rule' should be called 'rule violation' (a _scenario_ in a feature file in  gherkin -- gherkin groups tests by features containing scenarios)
- 'Rules' should be declared using the features names (e.g. 'Actual dates must be in the past')

We may have to do a bit of digging to see how to get the features names, as I'm not sure whether ```bdd-tester``` exposes them.

Ruleset results are output through either the web interface or the command line interface.

**Web interface**:

It would be nice to give the user the option of seeing the results pivoted in two different ways:

1 - rule -> [ rule violation -> [activity -> explanation, activity -> explanation, ..], rule violation -> ... ]

2-  activity - > [ rule -> [rule violation -> explanation, rule violation -> explanation], rule -> ... ]


**CLI**

We should ouput each ruleset violation like this:

```
    {
        'rule': feature name,
        'rule violation': sceneario ('instance' of feature) ,
        'explanation': message,
        'activity': activity
    }
```

Also, we discussed setting an URL to send a POST request with a data file to get a response with the same json output of the CLI.
